specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-06-02-preview/fleets.json

```diff
@@ -0,0 +1,817 @@
+        }
+      ],
+      "description": "The Fleet resource which contains multiple Kubernetes clusters as its members."
+    },
+    "FleetProperties": {
+      "type": "object",
+      "description": "Properties of a Fleet.",
+      "properties": {
+        "hubProfile": {
+          "description": "The FleetHubProfile configures the Fleet's hub.",
+          "$ref": "#/definitions/FleetHubProfile"
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "x-ms-enum": {
+            "name": "FleetProvisioningState",
+            "modelAsString": true
+          },
+          "enum": [
+            "Succeeded",
+            "Failed",
+            "Canceled",
+            "Creating",
+            "Deleting",
+            "Updating"
+          ],
+          "description": "The provisioning state of the last accepted operation."
+        }
+      }
+    },
+    "FleetHubProfile": {
+      "type": "object",
+      "description": "The FleetHubProfile configures the fleet hub.",
+      "properties": {
+        "dnsPrefix": {
+          "type": "string",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "description": "DNS prefix used to create the FQDN for the Fleet hub."
+        },
+        "fqdn": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The FQDN of the Fleet hub."
+        },
+        "kubernetesVersion": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The Kubernetes version of the Fleet hub."
+        }
+      }
+    },
+    "FleetListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Fleet"
+          },
+          "description": "The list of Fleets."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next page of Fleets.",
+          "readOnly": true
+        }
+      },
+      "description": "The response from the List Fleets operation."
+    },
+    "FleetMembersListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/FleetMember"
+          },
+          "description": "The list of members in a given Fleet."
+        },
+        "nextLink": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The URL to get the next page of Fleet members."
+        }
+      },
+      "description": "The response from the List FleetMembers operation."
+    },
+    "FleetPatch": {
+      "type": "object",
+      "description": "Properties of a Fleet that can be patched.",
+      "properties": {
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
```

Comment:
>additionalProperties [](http://example.com/codeflow?start=11&length=20)

Please spec the properties. Generally, the unschematized object pattern should only be used when needed and the object properties are dynamic, unknown, or user defined. Ref: [https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi032-only-use-additionalproperties-when-the-object-properties-are-dynamic-unknown-or-user-defined](https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi032-only-use-additionalproperties-when-the-object-properties-are-dynamic-unknown-or-user-defined).

Reply:
this is under tags. how do I schematize tags that are an arbitrary key-valuer pair map?
do you have an example?

Reply:
This is how tags are defined by the shared common types that we use in our swagger model :
https://github.com/Azure/azure-rest-api-specs/blob/5e7603d4591ae39f9c2cedea75c8d97185e0aab2/specification/common-types/resource-management/v3/types.json#L61

Here are examples of other resources using this exact schema for patching tags :

- [Databases](https://github.com/Azure/azure-rest-api-specs/blob/5e7603d4591ae39f9c2cedea75c8d97185e0aab2/specification/sql/resource-manager/Microsoft.Sql/preview/2021-11-01-preview/Databases.json#L1226)
- [Eventgrid Topics](https://github.com/Azure/azure-rest-api-specs/blob/5e7603d4591ae39f9c2cedea75c8d97185e0aab2/specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2021-10-15-preview/EventGrid.json#L11345)

Please advise on how to model it otherwise

Reply:
Sorry, you are correct. Tags are user-defined.